src/server/controllers/hotspot.js

```diff
@@ -0,0 +1,84 @@
+import httpStatus from 'http-status';
+
+import { respondWithError, respondWithSuccess } from '~/server/helpers/respond';
+import { signBooth } from '~/common/services/vote';
+import { isVotingBoothInitialized } from '~/common/services/contracts/booths';
+import Answers from '~/server/models/answer';
+import { getFromRedis, setInRedis } from '~/server/services/redis';
+
+async function read(req, res) {
+  const hotspotAddress = process.env.HOTSPOT_ADDRESS;
+  const hotspotPrivKey = process.env.HOTSPOT_PRIV_KEY;
+  const hotspotFestivalQuestion = parseInt(
+    process.env.HOTSPOT_FESTIVAL_QUESTION,
+  );
+  const hotspotMaxVotes = process.env.HOTSPOT_MAX_VOTES;
+
+  let nonce = await getFromRedis(`nonce:${hotspotAddress}`);
+  if (!nonce) {
+    nonce = 0;
+  } else {
+    nonce = parseInt(nonce);
+  }
+  // increment the nonce immediately to prevent nonce-collision
+  await setInRedis(`nonce:${hotspotAddress}`, nonce + 1);
```

Comment:
I believe this could easily lead to a race condition? Maybe using the [`INCR`](https://redis.io/commands/incr) command would work here? Or using a [transaction](https://redis.io/topics/transactions)?

Reply:
theoretically, yes, but in practice I think the likelihood of enough people standing outside furtherfield using this at the same time that this section of code overlaps is very small ... didn't know about INCR though, will swap that in since it definitely solves it